binderhub/build.py

```diff
@@ -93,6 +94,10 @@ def submit(self):
                 secret=client.V1SecretVolumeSource(secret_name=self.push_secret)
             ))
 
+        env = []
+        if self.git_credential:
+            env.append(client.V1EnvVar(name='GIT_CREDENTIAL_ENV', value=provider.git_credential))
```

Comment:
This should be `value=self.git_credential` rather than `value=provider.get_credential`